platform_plugin_communications/tasks.py

```diff
@@ -0,0 +1,168 @@
+"""
+Tasks module for platform_plugin_communications.
+"""
+import json
+import logging
+
+from celery import shared_task
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.utils.translation import gettext_noop
+
+from platform_plugin_communications.edxapp_wrapper.bulk_email import CourseEmail, _get_course_email_context
+from platform_plugin_communications.edxapp_wrapper.courseware_courses import get_course
+from platform_plugin_communications.edxapp_wrapper.instructor_tasks import (
+    InstructorTask,
+    queue_subtasks_for_query,
+    run_main_task,
+    send_course_email,
+)
+from platform_plugin_communications.target import Target
+
+User = get_user_model()
+
+log = logging.getLogger("edx.celery.task")
+
+
+@shared_task
+def send_bulk_course_email_to_learners(entry_id, _xblock_instance_args):
+    """
+    Send email to individual learners.
+
+    Extracted from: lms.djangoapps.instructor_task.tasks.send_bulk_course_email
+    """
+    action_name = gettext_noop("emailed")
+    visit_fcn = perform_delegate_email_batches_to_learners
+    return run_main_task(entry_id, visit_fcn, action_name)
+
+
+def perform_delegate_email_batches_to_learners(
+    entry_id, course_id, task_input, action_name
+):
+    """
+    Send email to individual learners in different subtasks.
```

Comment:
same question here about the teams modification. Also, why the `in different subtasks`?

Reply:
Subtasks are used to separate recipients into buckets of learners base on `settings.BULK_EMAIL_EMAILS_PER_TASK,` this allow us to send the emails in different tasks, improving times email delivery.

This is the same implementation as in edx-platform for `perform_delegate_email_batches`